Add GitHub Pages documentation

romandariana opened the pull request:
## Pull Request Description

Check out the result on my fork: https://romandariana.github.io/adi-kuiper-gen/ :D

Add docs folder with all needed source files for GitHub Pages documentation.

Add .github/workflows/top-level.yml to setup the GitHub Pages deployment workflow.

Move the complete documentation from README.md to GitHub Pages. Now README.md contains a succinct introduction and references the comprehensive documentation.

## PR Type
- [x] New feature (change that adds new functionality)

## PR Checklist
- [x] I have performed a self-review of the changes
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc)

gastmaier reviewed (approved):
Thank you :)

romandariana commented:
v2 -> v3:
    Added this page: https://romandariana.github.io/adi-kuiper-gen/hardware-configuration.html, documenting #211. The exact changes can be checked by viewing this commit on my fork: https://github.com/romandariana/adi-kuiper-gen/commit/ab1349a686bb7e8e54d438525795d2c0ea00c8ac. Here I squashed this commit together with the other changes.

gastmaier commented on .github/workflows/top-level.yml:
```diff
@@ -0,0 +1,81 @@
+on:
```
There is two new suggested yml file at 
https://analogdevicesinc.github.io/doctools/deploy.html#github-pages-deployment
that uses two new shared actions, `gh-pages-deploy@action` and `gh-pages-rm-path@action`.
The first deploys for commits in the main branch as usual, but also deploys for pull requests pull/<id> paths.
The second .yml is to then clean-up pull/<id> when it gets closed.

mlradu commented on docs/use-kuiper-image.rst:
```diff
@@ -0,0 +1,178 @@
+Writing the Image to an SD Card
+-------------------------------
+
+Using Balena Etcher (Recommended)
```
I would rather stick to the tools that we currently use so that we know that they were tested.

SRaus replied:
In my case, Balena Etcher is not working :)
I'm using Win32 Imager on Windows and Ubuntu Disk Image Writer or 'dd' on Linux.

gastmaier replied:
I believe balena etcher doesn't work on the corporate windows due to the special TI sauce.
We would need to test in a non-managed windows machine.

I agree that providing more tools options is better

mlradu commented on docs/build-flow.rst:
```diff
@@ -0,0 +1,171 @@
+
+.. code-block:: bash
+
+   sudo bash build-docker.sh
```
There are two options mentioned, with bash build-docker.sh and ./build-docker.sh. Even though the build-docker.sh was made executable, it can still be used with bash command.

SRaus replied:
`bash build-docker.sh` is redundant, switching to bash shell twice: first by using `bash...` in the command, and another time through shebang.

romandariana commented:
v4 -> v5:
    - implemented all suggestions (@SRaus) I marked as resolved (thank you!)
    - updated Repositories page according to how the packets are structured now
    - tried to respect the max 80 characters per line rule wherever it didn't ruin the format of the page
    
Exact changes: https://github.com/romandariana/adi-kuiper-gen/commit/0e23a388d16917d13ba1e68be5e5e29460b4a044. Result: https://analogdevicesinc.github.io/adi-kuiper-gen/pull/210/
